### ISSUE TITLE
🐛 [firebase_messaging] OnBackgroundMessage runs intermittently on iOS

### ISSUE THREAD

**Issue by saulrmzlpz** [firebase/flutterfire#11621]
## Bug report

**Describe the bug**
In some cases, the method for handling background notifications is not executed. Only if the app is opened at least once does it work. For example if the user deletes the app from recent and then restarts the phone.

### Steps to reproduce

1. Configure flutterfire/packages/firebase_messaging/example app with your own firebase parameters (GoogleService-Info.plist, etc.) 
2. Run app in release mode.
3. Copy token and send test notification as follow example:
```json
{
    "registration_ids": [
        "TOKEN_HERE"
    ],
    "priority": "high",
    "content_available": true,
    "notification": {
        "title": "Test Notification",
        "body": "Reset phone 1",
        "sound": "default",
        "content_available": true,
        "click_action": "FLUTTER_NOTIFICATION_CLICK"
    },
    "data": {
        "click_action": "FLUTTER_NOTIFICATION_CLICK",
        "type": "0",
        "url": "https://google.com.mx"
    },
    "apns": {
        "headers": {
            "apns-priority": "10"
        }
    }
}
``` 
4. Notification wil be received and background method works.
5. Kill app from recent and restart iPhone.
6. Send test notification again (Notification wil be received and background method works.)

### Expected behavior
As far as we understand the background process will start an isolated process in dart that will handle the actions that are scheduled with the constraints. (Our case, saving the notifications in sqlite). But for the case where the app has not been opened for a long period of time or the device has been rebooted, we will lose these functions.

### Sample project
The same project of this repository
---

## Additional context
Xcode 15.0
iOS 17.0 iPhone 13 Pro Max
---

### Flutter doctor

<details><summary>Click To Expand</summary>

```
[✓] Flutter (Channel stable, 3.13.4, on macOS 13.5.2 22G91 darwin-x64, locale es-419)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.3)
[✓] VS Code (version 1.82.2)
[✓] Connected device (3 available)
[✓] Network resources
```

</details>

---

### Flutter dependencies

<details><summary>Click To Expand</summary>

```
- _flutterfire_internals 1.3.6 [collection firebase_core firebase_core_platform_interface flutter meta]
- args 2.4.2
- async 2.11.0 [collection meta]
- boolean_selector 2.1.1 [source_span string_scanner]
- characters 1.3.0
- clock 1.1.1
- collection 1.17.2
- dbus 0.7.8 [args ffi meta xml]
- fake_async 1.3.1 [clock collection]
- ffi 2.1.0
- firebase_core_platform_interface 4.8.0 [collection flutter flutter_test meta plugin_platform_interface]
- firebase_core_web 2.8.0 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_messaging_platform_interface 4.5.7 [_flutterfire_internals firebase_core flutter meta plugin_platform_interface]
- firebase_messaging_web 3.5.7 [_flutterfire_internals firebase_core firebase_core_web firebase_messaging_platform_interface flutter flutter_web_plugins js meta]
- flutter_local_notifications_linux 4.0.0+1 [dbus ffi flutter flutter_local_notifications_platform_interface path xdg_directories]
- flutter_local_notifications_platform_interface 7.0.0+1 [flutter plugin_platform_interface]
- flutter_test 0.0.0 [flutter test_api matcher path fake_async clock stack_trace vector_math async boolean_selector characters collection material_color_utilities meta source_span stream_channel string_scanner term_glyph web]
- flutter_web_plugins 0.0.0 [flutter characters collection material_color_utilities meta vector_math web]
- http_parser 4.0.2 [collection source_span string_scanner typed_data]
- js 0.6.7 [meta]
- matcher 0.12.16 [async meta stack_trace term_glyph test_api]
- material_color_utilities 0.5.0 [collection]
- meta 1.9.1
- path 1.8.3
- petitparser 5.4.0 [meta]
- sky_engine 0.0.99
- source_span 1.10.0 [collection path term_glyph]
- stack_trace 1.11.0 [path]
- stream_channel 2.1.1 [async]
- string_scanner 1.2.0 [source_span]
- term_glyph 1.2.1
- test_api 0.6.0 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph]
- timezone 0.9.2 [path]
- typed_data 1.3.2 [collection]
- vector_math 2.1.4
- web 0.1.4-beta
- xdg_directories 1.0.3 [meta path]
- xml 6.3.0 [collection meta petitparser]
```

</details>

---

**Comment by JYeop** [comment]
Same thing happens here.
When the app killed & receives background fcm, first message gets well.
But the second fcm not arrives, server says "Request contains an invalid argument". (Golang + firebase admin sdk)
But i looked the payload, it was totally fine everytime.
Then i launched the app, fcm token has changed. (FirebaseMessaging.instance.getToken())
It is strange bcs the token always changes when the flow happens. (app killed -> background message -> open app).
I did not executed "deleteToken()".

In summary,

Only the first background message arrives when the app is killed, cannot receive next messages.
After the first background message receiving, somehow the fcm token always changed

**Comment by pmatatias** [comment]
In my case, `onBackgroundMessageHanlder` never invoked for `iOS` device. 
I receive the notification, but the method was not invoke at all

Honestly im bit **confused**, the documentation said that :
src: https://firebase.flutter.dev/docs/messaging/overview#4-apple-integration

> Firebase Messaging plugin will not work if you disable method swizzling. Please remove or set to true the FirebaseAppDelegateProxyEnabled in your Info.plist file if it exists.

Im make sure that my `info.plist` not contain `FirebaseAppDelegateProxyEnabled`. But the backgorund handler never invoked.

----
then i fix it by this workaround from stackoverflow:
src: https://stackoverflow.com/a/38445657/12838877

> Add FirebaseAppDelegateProxyEnabled type "Boolean" value "NO" in your info.plist

after add this to my `info.plist`, finally my `backgroundHandler` was invoked.

**Comment by danagbemava-nc** [comment]
Hi @saulrmzlpz, this is the documented behavior for background messages on iOS. See the preconditions section of this document https://firebase.google.com/docs/cloud-messaging/flutter/receive. 

Is there anything that isn't clear?

**Comment by saulrmzlpz** [comment]
> Similar thing happens here. Im sending fcm data message. When the app killed & receives background fcm, first message gets well. But the second fcm not arrives, server says "Request contains an invalid argument". (Golang + firebase admin sdk) I looked the payload, it was totally fine everytime. Then i launched the app, fcm token has changed. (FirebaseMessaging.instance.getToken()) It is strange bcs the token always changes when the flow happens. (app killed -> background message -> open app). I did not executed "deleteToken()".
> 
> In summary,
> 
> 1. Only the first background message arrives when the app is killed, cannot receive next messages.
> 2. After the first background message receiving, somehow the fcm token always changed



> Similar thing happens here. Im sending fcm data message. When the app killed & receives background fcm, first message gets well. But the second fcm not arrives, server says "Request contains an invalid argument". (Golang + firebase admin sdk) I looked the payload, it was totally fine everytime. Then i launched the app, fcm token has changed. (FirebaseMessaging.instance.getToken()) It is strange bcs the token always changes when the flow happens. (app killed -> background message -> open app). I did not executed "deleteToken()".
> 
> In summary,
> 
> 1. Only the first background message arrives when the app is killed, cannot receive next messages.
> 2. After the first background message receiving, somehow the fcm token always changed

@JYeop  Check my recent closed issue https://github.com/firebase/flutterfire/issues/11618

**Comment by danagbemava-nc** [comment]
Hi @saulrmzlpz, are you now clear on the behavior of background messages on iOS?

**Comment by saulrmzlpz** [comment]
> Hi @saulrmzlpz, are you now clear on the behavior of background messages on iOS?

Hi. There is no problem since what is reported is part of the preconditions mentioned in the documentation
